app/services/SubscriptionFieldsService.scala

```diff
@@ -57,9 +58,9 @@ class SubscriptionFieldsService @Inject()(@Named("SANDBOX") sandboxSubscriptionF
     connectorFor(application).fetchFieldsValuesWithPrefetchedDefinitions(application.clientId, apiIdentifier, definitions)
   }
 
-  def saveFieldValues(application: Application, apiContext: ApiContext, apiVersion: ApiVersion, fields: Fields.Alias)
+  def saveFieldValues(application: NewApplication, apiContext: ApiContext, apiVersion: ApiVersion, fields: Fields.Alias)
       (implicit hc: HeaderCarrier): Future[SaveSubscriptionFieldsResponse] = {
-    connectorFor(application).saveFieldValues(application.clientId, apiContext, apiVersion, fields)
+    connectorFor(application.deployedTo.toString).saveFieldValues(application.clientId, apiContext, apiVersion, fields)
```

Comment:
Could we make this method use the case class ?

Reply:
Not yet - other places that use it only have a String.